src/Microsoft.ComponentDetection.Detectors/poetry/PoetryComponentDetector.cs

```diff
@@ -54,17 +54,15 @@ protected override async Task OnFileFoundAsync(ProcessRequest processRequest, ID
 
         poetryLock.Package.ToList().ForEach(package =>
         {
-            var isDevelopmentDependency = package.Category != "main";
```

Comment:
For versions of poetry >=1.5.0, the category field is not written to the lockfile since poetry has replaced the concept of categories.

Like cargo/rust, it's not longer possible to identify that a dependency is solely a dev dependency from the lockfile alone. It could be deduced from a combination of the pyproject.toml and the poetry.lock but I wanted to avoid doing that for the same rationale as https://github.com/microsoft/component-detection/pull/117 - it's fragile and could result in false negatives.

I didn't bother retaining support for dev dependency detection from older lockfiles, as 1.5.0 is nearly a year old now which is quite a long time for poetry (and newer python packaging tools which are evolving quickly).

Reply:
I think that's a fair assessment of the limitations of using the lock file parsing here. I've opened #996 to investigate future improvements here.